schemas/2016-04-30-preview/Microsoft.Compute.json

```diff
@@ -0,0 +1,4781 @@
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ]
+        },
+        "resources": {
+          "type": "array",
+          "items": {
+            "oneOf": [
+              {
+                "$ref": "#/resourceDefinitions/extensionsChild"
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "type",
+        "apiVersion",
+        "properties",
+        "location"
+      ],
+      "description": "Microsoft.Compute/virtualMachines"
+    },
+    "extensions": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "Microsoft.Compute/virtualMachines/extensions"
+          ]
+        },
+        "apiVersion": {
+          "enum": [
+            "2016-04-30-preview"
+          ]
+        },
+        "properties": {
+          "anyOf": [
+            { "$ref": "#/definitions/genericExtension" },
+            { "$ref": "#/definitions/iaaSDiagnostics" },
+            { "$ref": "#/definitions/iaaSAntimalware" },
+            { "$ref": "#/definitions/customScriptExtension" },
+            { "$ref": "#/definitions/customScriptForLinux"},
+            { "$ref": "#/definitions/linuxDiagnostic"},
+            { "$ref": "#/definitions/vmAccessForLinux"},
+            { "$ref": "#/definitions/bgInfo"},
+            { "$ref": "#/definitions/vmAccessAgent"},
+            { "$ref": "#/definitions/dscExtension"},
+            { "$ref": "#/definitions/acronisBackupLinux"},
+            { "$ref": "#/definitions/acronisBackup"},
+            { "$ref": "#/definitions/linuxChefClient"},
+            { "$ref": "#/definitions/chefClient"},
+            { "$ref": "#/definitions/datadogLinuxAgent"},
+            { "$ref": "#/definitions/datadogWindowsAgent"},
+            { "$ref": "#/definitions/dockerExtension"},
+            { "$ref": "#/definitions/dynatraceLinux"},
+            { "$ref": "#/definitions/dynatraceWindows"},
+            { "$ref": "#/definitions/eset"},
+            { "$ref": "#/definitions/hpeSecurityApplicationDefender"},
+            { "$ref": "#/definitions/puppetAgent"},
+            { "$ref": "#/definitions/site24x7LinuxServerExtn"},
+            { "$ref": "#/definitions/site24x7WindowsServerExtn"},
+            { "$ref": "#/definitions/site24x7ApmInsightExtn"},
+            { "$ref": "#/definitions/trendMicroDSALinux"},
+            { "$ref": "#/definitions/trendMicroDSA"},
+            { "$ref": "#/definitions/bmcCtmAgentLinux"},
+            { "$ref": "#/definitions/bmcCtmAgentWindows"},
+            { "$ref": "#/definitions/OSPatchingForLinux"},
+            { "$ref": "#/definitions/VMSnapshot"},
+            { "$ref": "#/definitions/VMSnapshotLinux"},
+            { "$ref": "#/definitions/customScript" }
+          ]
+        }
+      },
+      "required": [
+        "type",
+        "properties",
+        "apiVersion"
```

Comment:
There's a recent change which adds location in the required list. You can sync master and get that.